olive/engine/engine.py

```diff
@@ -128,28 +129,51 @@ def initialize(self):
 
         # clean pass run cache if requested
         # removes all run cache for pass type and all children elements
-        for pass_name in self.pass_order:
-            clean_run_cache = self.passes[pass_name]["clean_run_cache"]
-            p = self.passes[pass_name]["pass"]
+        for pass_name, pass_config in self.pass_config.items():
+            clean_run_cache = pass_config["clean_run_cache"]
             if clean_run_cache:
-                cache_utils.clean_pass_run_cache(p.__class__.__name__, cache_dir)
-
-        # list of passes starting from the first pass with non-empty search space
-        # These passes will be added to the search space
-        self.pass_search_spaces = []
-        for pass_name in self.pass_order:
-            p = self.passes[pass_name]["pass"]
-            self.pass_search_spaces.append((pass_name, p.search_space()))
+                cache_utils.clean_pass_run_cache(pass_name, cache_dir)
```

Comment:
`pass_name` is not always the same as the name of the pass class. It can be user provided too. It is always user provided for workflow runs. 
Can you make this to `p.__name__`  instead?

Reply:
so, we'd like to use the pass type name to clean the cache instead of using the user provided pass name. Is that correct?